universal-application-tool-0.0.1/app/services/question/QuestionDefinitionBuilder.java

```diff
@@ -56,18 +56,20 @@ public QuestionDefinitionBuilder setQuestionType(QuestionType questionType) {
     return this;
   }
 
-  public QuestionDefinition build() {
+  public QuestionDefinition build() throws UnsupportedQuestionTypeException {
     switch (this.questionType) {
       case ADDRESS:
         return new AddressQuestionDefinition(
             id, version, name, path, description, questionText, questionHelpText);
       case NAME:
         return new NameQuestionDefinition(
             id, version, name, path, description, questionText, questionHelpText);
-      case TEXT: // fallthrough intended.
-      default:
-        return new QuestionDefinition(
+      case TEXT:
+        return new TextQuestionDefinition(
             id, version, name, path, description, questionText, questionHelpText);
+      default:
+        // TODO: throw a real exception.
```

Comment:
nit: seems like you implemented this TODO and it can be removed

Reply:
Conversation was resolved while the TODO is still there?

Reply:
i might be having git troubles. The tests are failing on origin but look fine on my end.

Reply:
i might be having git troubles. The tests are failing on origin but look fine on my end. when i try to push it says everything is up-to-date.